src/utils/vectorEncoder.ts

```diff
@@ -0,0 +1,398 @@
+          encodNumber = differenceDigit(diffData[index].toString());
+          if (encodNumber.length + charCount < MaxLinelength) {
+            encodData += encodNumber;
+            charCount += encodNumber.length;
+          } else {
+            // Iniciar nueva linea
+            encodData += newLine;
+            temp =
+              Math.ceil(firstX + index * intervalX) +
+              squeezedDigit(data[index].toString()) +
+              encodNumber;
+            encodData += temp; // Each line start with first index number.
+            charCount = temp.length;
+          }
+        }
+      }
+    }
+    index++;
+  }
+  if (mult > 0) {
+    encodData += duplicateDigit((mult + 1).toString());
+  }
+  // We insert the last data from fid. It is done to control of data
+  // The last line start with the number of datas in the fid.
+  encodData +=
+    newLine +
+    Math.ceil(firstX + index * intervalX) +
+    squeezedDigit(data[index].toString());
+
+  return encodData;
+}
+
+/**
+ * @private
+ * Differential encoding
+ * @param {Array} data
+ * @param {number} firstX
+ * @param {number} intervalX
+ * @return {string}
+ */
+export function differenceEncoding(
+  data: DoubleArray,
+  firstX: number,
+  intervalX: number,
+) {
+  let index = 0;
+  let charCount = 0;
+  let i;
+
+  let encodData = '';
+  let encodNumber = '';
+  let temp = '';
+
+  // We calculate the differences vector
+  let diffData = new Array(data.length - 1);
+  for (i = 0; i < diffData.length; i++) {
+    diffData[i] = data[i + 1] - data[i];
+  }
+
+  let numDiff = diffData.length;
+  while (index < numDiff) {
+    if (charCount === 0) {
+      // We convert the first number.
+      encodNumber =
+        Math.ceil(firstX + index * intervalX) +
+        squeezedDigit(data[index].toString()) +
+        differenceDigit(diffData[index].toString());
+      encodData += encodNumber;
+      charCount += encodNumber.length;
+    } else {
+      encodNumber = differenceDigit(diffData[index].toString());
+      if (encodNumber.length + charCount < MaxLinelength) {
+        encodData += encodNumber;
+        charCount += encodNumber.length;
+      } else {
+        encodData += newLine;
+        temp =
+          Math.ceil(firstX + index * intervalX) +
+          squeezedDigit(data[index].toString()) +
+          encodNumber;
+        encodData += temp; // Each line start with first index number.
+        charCount = temp.length;
+      }
+    }
+    index++;
+  }
+  // We insert the last number from data. It is done to control of data
+  encodData +=
+    newLine +
+    Math.ceil(firstX + index * intervalX) +
+    squeezedDigit(data[index].toString());
+
+  return encodData;
+}
+
+/**
+ * @private
+ * Convert number to the ZQZ format, using pseudo digits.
+ * @return {string}
+ */
+function squeezedDigit(num: string) {
+  let SQZdigit = '';
+  if (num.startsWith('-')) {
+    SQZdigit += pseudoDigits[SQZ_N][Number(num.charAt(1))];
```

Comment:
To improve speed you don't need to convert a string to a number but just convert it directly from the ascii code.

Something like `'a12'.charCodeAt(1)-65;` and you need to check to have the right shift value. This change should be done everything you have a charAt I think